v0/destinations/snapchat_conversion/transform.js

```diff
@@ -185,27 +186,50 @@ function trackResponseBuilder(message, { Config }) {
     payload.event_tag = message.properties.event_tag;
   }
 
-  payload.hashed_email = getHashedValue(
-    message?.context?.traits?.email
-      ?.trim()
-      ?.toString()
-      ?.toLowerCase()
-  );
-  payload.hashed_phone_number = getHashedValue(
-    getNormalizedPhoneNumber(message)
-      ?.toString()
-      ?.toLowerCase()
-  );
-  payload.user_agent = message?.context?.userAgent?.toString()?.toLowerCase();
-  payload.hashed_ip_address = getHashedValue(
-    message?.context?.ip?.toString()?.toLowerCase()
-  );
-  (hashed_mobile_ad_id = getHashedValue(
-    message?.context?.idfa?.toString()?.toLowerCase()
-  )),
-    (hashed_idfv = getHashedValue(
-      message?.context?.idfv?.toString()?.toLowerCase()
-    ));
+  const email = getFieldValueFromMessage(message, "email");
+  if (email) {
+    payload.hashed_email = getHashedValue(
+      email
+        .toString()
+        .toLowerCase()
+        .trim()
+    );
+  }
+  const phone = getNormalizedPhoneNumber(message);
+  if (phone) {
+    payload.hashed_phone_number = getHashedValue(
+      phone
+        .toString()
+        .toLowerCase()
+        .trim()
+    );
+  }
+  const ip = message.context?.ip || message.request_ip;
+  if (ip) {
+    payload.hashed_ip_address = getHashedValue(
+      ip
+        .toString()
+        .toLowerCase()
+        .trim()
+    );
+  }
+  // only in case of ios platform this is required
+  if (
+    isAppleFamily(message.context?.device?.type) &&
+    (message.properties?.idfv || message.contex?.device?.id)
+  ) {
+    payload.hashed_idfv = getHashedValue(
+      message.properties?.idfv || message.contex?.device?.id
+    );
+  }
+
+  if (message.properties?.adId || message.context?.device?.advertisingId) {
```

Comment:
Does Snapchat conversion have a use case when even if the device type is of the apple family it takes say android related data? advertisingId (ad_id) is mostly for anonymous tacking on android devices.

Reply:
`idfv` is for apple only, its mentioned in Snapchat docs

`ad_id` is not mentioned for specific for any device type